src/basicfuns.jl

```diff
@@ -224,14 +224,14 @@ logsubexp(x::Real, y::Real) = max(x, y) + log1mexp(-abs(x - y))
 """
     logsumexp(X)
 
-Compute `log(sum(exp, X))`, evaluated avoiding intermediate overflow/undeflow.
+Compute `log(sum(exp, X))` in a numerically stable way that avoids intermediate over- and
+underflow.
 
 `X` should be an iterator of real numbers.
+
+See also: [`logsumexp_onepass`](@ref)
 """
-function logsumexp(X)
-    isempty(X) && return log(sum(X))
-    reduce(logaddexp, X)
-end
+logsumexp(X) = logsumexp_onepass(X)
 function logsumexp(X::AbstractArray{T}; dims=:) where {T<:Real}
```

Comment:
When `dims=:`,  `logsumexp_onepass` should be called. The standard way to achieve this is to have `_logsumexp(X, ::Colon)` and `_logsumexp(X::AbstractArray, dims)`, and call them from `logsumexp(X; dims=:) = _logsumexp(X, dims)`.

Also there's no need to define `logsumexp_onepass`, just make it a `_logsumexp` method.